app/code/community/Bolt/Boltpay/Model/Quote.php

```diff
@@ -246,13 +246,13 @@ function ($item) use ($quote, &$calculatedTotal, $boltHelper) {
                     'first_name'      => $shippingAddress->getFirstname(),
                     'last_name'       => $shippingAddress->getLastname(),
                     'locality'        => $shippingAddress->getCity(),
-                    'region'          => $region,
-                    'postal_code'     => $shippingAddress->getPostcode(),
+                    'region'          => $shippingRegion,
+                    'postal_code'     => $shippingAddress->getPostcode() ?: '-',
                     'country_code'    => $shippingAddress->getCountry(),
                     'phone'           => $shippingAddress->getTelephone(),
-                    'email'           => $shippingAddress->getEmail() ?: $customerEmail,
+                    'email'           => $shippingAddress->getEmail() ?: ($customerEmail ?: $billingAddress->getEmail()),
                     'phone_number'    => $shippingAddress->getTelephone(),
-                    'email_address'   => $shippingAddress->getEmail() ?: $customerEmail,
+                    'email_address'   => $shippingAddress->getEmail() ?: ($customerEmail ?: $billingAddress->getEmail()),
                 );
```

Comment:
The doubling of fields, phone / phone_number and email / email_address, is a relic of the past when at some point the names were changing back and forth on the Bolt server side. The line of communication was not so well established at the time so this was my solution to it. I think phone_number and email_address are redundant now, at least they are not present in Magento2 plugin create order request.

Reply:
@danac-gs I'm aware of this.  Both those fields are undocumented and with confirmation from @daisy1754, I will remove those fields.

Reply:
can confirm these are no longer used